docs/_docs/02_features/stacks.md

```diff
@@ -0,0 +1,536 @@
+
+```hcl
+dependency "vpc" {
+  config_path = "../vpc"
+
+  mock_outputs = {
+    vpc_id = "temporary-dummy-id"
+  }
+
+  skip_outputs = true
+}
+```
+
+You can also use `mock_outputs_merge_strategy_with_state` on the `dependency` block to merge mocked outputs and real outputs:
+
+```hcl
+dependency "vpc" {
+  config_path = "../vpc"
+
+  mock_outputs = {
+    vpc_id     = "temporary-dummy-id"
+    new_output = "temporary-dummy-value"
+  }
+
+  mock_outputs_merge_strategy_with_state = "shallow"
+}
+```
+
+If real outputs only contain `vpc_id`, `dependency.outputs` will contain a real value for `vpc_id` and a mocked value for `new_output`.
+
+## Dependencies between units
+
+You can also specify dependencies explicitly. Consider the following file structure:
+
+```tree
+root
+├── backend-app
+│   ├── main.tf
+│   └── terragrunt.hcl
+├── frontend-app
+│   ├── main.tf
+│   └── terragrunt.hcl
+├── mysql
+│   ├── main.tf
+│   └── terragrunt.hcl
+├── redis
+│   ├── main.tf
+│   └── terragrunt.hcl
+└── vpc
+    ├── main.tf
+    └── terragrunt.hcl
+```
+
+Let’s assume you have the following dependencies between OpenTofu/Terraform units:
+
+- `backend-app` depends on `mysql`, `redis`, and `vpc`
+
+- `frontend-app` depends on `backend-app` and `vpc`
+
+- `mysql` depends on `vpc`
+
+- `redis` depends on `vpc`
+
+- `vpc` has no dependencies
+
+You can express these dependencies in your `terragrunt.hcl` config files using a `dependencies` block. For example, in `backend-app/terragrunt.hcl` you would specify:
+
+``` hcl
+dependencies {
+  paths = ["../vpc", "../mysql", "../redis"]
+}
+```
+
+Similarly, in `frontend-app/terragrunt.hcl`, you would specify:
+
+``` hcl
+dependencies {
+  paths = ["../vpc", "../backend-app"]
+}
+```
+
+Once you’ve specified these dependencies in each `terragrunt.hcl` file, Terragrunt will be able to perform updates respecting the [DAG](/docs/getting-started/terminology/#directed-acyclic-graph-dag) of dependencies.
+
+For the example at the start of this section, the order of runs for the `run-all apply` command would be:
+
+1. Deploy the VPC
+
+2. Deploy MySQL and Redis in parallel
+
+3. Deploy the backend-app
+
+4. Deploy the frontend-app
+
+Any error encountered in an individual unit during a `run-all` command will prevent Terragrunt from proceeding with the deployment of any dependent units.
+
+To check all of your dependencies and validate the code in them, you can use the `run-all validate` command.
+
+**Note:** During `destroy` runs, Terragrunt will try to find all dependent units and show a confirmation prompt with a list of detected dependencies. This is because Terragrunt knows that once resources in a dependency is destroyed, any commands run on dependent units may fail. For example, if `destroy` was called on the `redis` unit, you'll be asked for confirmation, as the `backend-app` depends on `redis`. You can avoid the prompt by using `--terragrunt-non-interactive`.
+
+## Visualizing the DAG
```

Comment:
is DAG a universal enough acronym to use without nearby definition?

Reply:
I referenced the entry in the terminology up above:
https://github.com/gruntwork-io/terragrunt/pull/3640/files/27bd5c1209ad7ac2ca79dd42764ba605da2efe6d#diff-06b0593a1f8fd955ca3da66bc777edf4baf32a5c821014ca7bd42e74786b88b6R179

Do you think I'm better off linking to it everywhere?